packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx

```diff
@@ -185,6 +183,25 @@ describe('<DataGrid /> - Layout & Warnings', () => {
         }).toWarnDev(["You are calling getValue('age') but the column `age` is not defined"]);
         expect(getColumnValues()).to.deep.equal(['1', '2']);
       });
+
+      it('should have a stable height if the parent container has no intrinsic height', () => {
+        expect(() => {
+          const { getByRole } = render(
+            <div>
+              <p>The table keeps growing... and growing...</p>
+              <DataGrid {...baselineProps} />
+            </div>,
+          );
+          const firstHeight = getByRole('grid').clientHeight;
+          // Use timeout to allow simpler tests in JSDOM.
+          clock.tick(0);
+          const secondHeight = getByRole('grid').clientHeight;
+          expect(firstHeight).to.equal(secondHeight);
+          // @ts-expect-error need to migrate helpers to TypeScript
+        }).toWarnDev(
+          'Material-UI: useResizeContainer - The parent of the grid has an empty height.',
+        );
+      });
     });
```

Comment:
I have tried the test in HEAD with this diff:

```diff
diff --git a/packages/grid/_modules_/grid/components/AutoSizer.tsx b/packages/grid/_modules_/grid/components/AutoSizer.tsx
index 139d6d9f..fe435ce2 100644
--- a/packages/grid/_modules_/grid/components/AutoSizer.tsx
+++ b/packages/grid/_modules_/grid/components/AutoSizer.tsx
@@ -130,7 +130,7 @@ export const AutoSizer = React.forwardRef<HTMLDivElement, AutoSizerProps>(functi
   const childParams: any = {};

   if (!disableHeight) {
-    outerStyle.height = 0;
+    outerStyle.height = 'auto';
     childParams.height = state.height;
   }

diff --git a/packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx b/packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx
index 22cf2769..2a8871b7 100644
--- a/packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx
+++ b/packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx
@@ -185,6 +185,20 @@ describe('<DataGrid /> - Layout & Warnings', () => {
         }).toWarnDev(["You are calling getValue('age') but the column `age` is not defined"]);
         expect(getColumnValues()).to.deep.equal(['1', '2']);
       });
+
+      it.only('should have a stable height if the parent container has no intrinsic height', () => {
+        const { getByRole } = render(
+          <div>
+            <p>The table keeps growing... and growing...</p>
+            <DataGrid {...baselineProps} />
+          </div>,
+        );
+        const firstHeight = getByRole('grid').clientHeight;
+        // Use timeout to allow simpler tests in JSDOM.
+        clock.tick(0);
+        const secondHeight = getByRole('grid').clientHeight;
+        expect(firstHeight).to.equal(secondHeight);
+      });
     });

     describe('column width', () => {
```

It's not failing, so the test is not correct. It needs a real `sleep()` call, a real setTimeout, not the fake one `useFakeTimers` creates. Instead, I have updated the tests to:

```diff
diff --git a/packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx b/packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx
index 22cf2769..09f21a2a 100644
--- a/packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx
+++ b/packages/grid/data-grid/src/tests/layout.DataGrid.test.tsx
@@ -8,7 +8,7 @@ import {
 import { useFakeTimers } from 'sinon';
 import { expect } from 'chai';
 import { DataGrid, ValueGetterParams } from '@material-ui/data-grid';
-import { getColumnValues, raf } from 'test/utils/helperFn';
+import { getColumnValues, raf, sleep } from 'test/utils/helperFn';

 describe('<DataGrid /> - Layout & Warnings', () => {
   // TODO v5: replace with createClientRender
@@ -187,6 +187,20 @@ describe('<DataGrid /> - Layout & Warnings', () => {
       });
     });

+    it.only('should have a stable height if the parent container has no intrinsic height', async () => {
+      const { getByRole } = render(
+        <div>
+          <p>The table keeps growing... and growing...</p>
+          <DataGrid {...baselineProps} />
+        </div>,
+      );
+      const firstHeight = getByRole('grid').clientHeight;
+      // Use timeout to allow simpler tests in JSDOM.
+      await sleep(10);
+      const secondHeight = getByRole('grid').clientHeight;
+      expect(firstHeight).to.equal(secondHeight);
+    });
+
     describe('column width', () => {
       it('should set the columns width to 100px by default', () => {
         const rows = [
```

Now, it fails, which is correct, the condition is met:

<img width="428" alt="Capture d’écran 2021-01-29 à 20 40 02" src="https://user-images.githubusercontent.com/3165635/106319791-48140f00-6272-11eb-8939-9c8959582e85.png">

It grows infinitely it evens log 

> Warning: `Infinity` is an invalid value for the `minHeight` css style property.

I'm pushing a new commit.